crates/bitwarden-auth/src/registration.rs

```diff
@@ -33,4 +49,99 @@ impl RegistrationClient {
+    ) -> Result<KeyConnectorRegistrationResult, UserRegistrationError> {
+        let client = &self.client.internal;
+        let api_client = &client.get_api_configurations().await.api_client;
+        let user_id =
+            UserId::from_str(user_id.as_str()).map_err(|_| UserRegistrationError::Serialization)?;
+
+        // First call crypto API to get all keys
+        info!("Initializing account cryptography");
+        let (
+            cryptography_state,
+            wrapped_user_key,
+            user_key,
+            account_cryptographic_state_request,
+            key_connector_key,
+        ) = self
+            .client
+            .crypto()
+            .make_user_key_connector_registration(user_id)
+            .map_err(UserRegistrationError::AccountCryptographyMakeKeys)?;
+
+        info!("Posting key connector key to key connector server");
+        key_connector_api_post_or_put_key_connector_key(
+            &self.client,
+            key_connector_url.as_str(),
+            &key_connector_key,
+        )
+        .await
+        .map_err(UserRegistrationError::KeyConnectorApi)?;
+
+        info!("Posting user account cryptographic state to server");
+        let request = SetKeyConnectorKeyRequestModel {
+            key_connector_key_wrapped_user_key: Some(wrapped_user_key.to_string()),
+            account_keys: Some(Box::new(account_cryptographic_state_request)),
+            ..SetKeyConnectorKeyRequestModel::new(org_id)
+        };
+        api_client
+            .accounts_key_management_api()
+            .post_set_key_connector_key(Some(request))
+            .await
+            .map_err(|e| UserRegistrationError::Api(e.into()))?;
+
+        info!("User initialized!");
+        // Note: This passing out of state and keys is temporary. Once SDK state management is more
+        // mature, the account cryptographic state and keys should be set directly here.
+        Ok(KeyConnectorRegistrationResult {
+            account_cryptographic_state: cryptography_state,
+            key_connector_key: key_connector_key.to_base64(),
+            key_connector_key_wrapped_user_key: wrapped_user_key,
+            user_key: user_key.to_encoded().to_vec().into(),
+        })
+    }
+}
+
+/// Result of Key Connector registration process.
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)]
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct KeyConnectorRegistrationResult {
+    /// The account cryptographic state of the user.
+    pub account_cryptographic_state: WrappedAccountCryptographicState,
+    /// The key connector key used for unlocking.
+    pub key_connector_key: B64,
+    /// The encrypted user key, wrapped with the key connector key.
+    pub key_connector_key_wrapped_user_key: EncString,
+    /// The decrypted user key. This can be used to get the consuming client to an unlocked state.
+    pub user_key: ByteBuf,
```

Comment:
This should be updated to B64 (the TDE PR does this too), since both user_key and key_connector_key are keys. Also, ByteBuf is not nicely supported by uniffi